buildSrc/src/main/kotlin/io/spine/internal/gradle/publish/JarDsl.kt

```diff
@@ -0,0 +1,167 @@
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.spine.internal.gradle.publish
+
+/**
+ * A DSL element of [SpinePublishing] extension which configures publishing of
+ * [dokkaKotlinJar] artifact.
+ *
+ * This artifact contains Dokka-generated documentation. By default, it is not published.
+ *
+ * Take a look at the [SpinePublishing.dokkaJar] for a usage example.
+ *
+ * @see [artifacts]
+ */
+class DokkaJar {
+    /**
+     * Enables publishing `JAR`s with Dokka-generated documentation for all published modules.
+     */
+    @Suppress("unused")
+    @Deprecated("Please use `kotlin` and `java` flags instead.")
+    var enabled = false
+
+    /**
+     * Controls whether [dokkaKotlinJar] artifact should be published.
+     * The default value is `true`.
+     */
+    var kotlin = true
+
+    /**
+     * Controls whether [dokkaJavaJar] artifact should be published.
+     * The default value is `false`.
+     */
+    var java = false
+}
+
+/**
+ * A DSL element of [SpinePublishing] extension which allows enabling publishing
+ * of [testJar] artifact.
+ *
+ * This artifact contains compilation output of `test` source set. By default, it is not published.
+ *
+ * Take a look on [SpinePublishing.testJar] for a usage example.
+
+ * @see [artifacts]
+ */
+class TestJar {
+
+    /**
+     * Set of modules, for which a test JAR will be published.
+     */
+    var inclusions: Set<String> = emptySet()
+
+    /**
+     * Enables test JAR publishing for all published modules.
+     */
+    var enabled = false
+}
+
+/**
+ * A DSL element of [SpinePublishing] extension which allows disabling publishing
+ * of [protoJar] artifact.
+ *
+ * This artifact contains all the `.proto` definitions from `sourceSets.main.proto`. By default,
+ * it is published.
+ *
+ * Take a look on [SpinePublishing.protoJar] for a usage example.
+ *
+ * @see [artifacts]
+ */
+class ProtoJar {
+
+    /**
+     * Set of modules, for which a proto JAR will not be published.
+     */
+    var exclusions: Set<String> = emptySet()
+
+    /**
+     * Disables proto JAR publishing for all published modules.
+     */
+    var disabled = false
+}
+
+/**
+ * Flags for turning optional JAR artifacts in a project.
+ */
+internal data class JarFlags(
+
+    /**
+     * Tells whether [sourcesJar] artifact should be published.
+     *
+     * Default value is `true`.
+     */
+    val sourcesJar: Boolean = true,
+
+    /**
+     * Tells whether [javadocJar] artifact should be published.
+     *
+     * Default value is `true`.
+     */
+    val javadocJar: Boolean = true,
+    
+    /**
+     * Tells whether [protoJar] artifact should be published.
```

Comment:
If we started by explicitly describing the defaults for the previous fields, I say we continue here and below.